src/GafferSceneUI/Inspector.cpp

```diff
@@ -231,19 +278,57 @@ void Inspector::inspectHistoryWalk( const GafferScene::SceneAlgo::History *histo
 
 					if( !result->m_editScope || node->ancestor<EditScope>() == result->m_editScope )
 					{
-						const GraphComponent *readOnlyReason = MetadataAlgo::readOnlyReason( result->m_source.get() );
-						if( !result->m_editScope && readOnlyReason )
+						std::string nonEditableReason;
+						if( auto n = runTimeCast<NameValuePlug>( result->m_source.get() ) )
 						{
-							result->m_editFunction = fmt::format(
-								"{} is locked.",
-								readOnlyReason->relativeName( readOnlyReason->ancestor<ScriptNode>() )
+							nonEditableReason = lockedReason( n, result->m_editScope.get() );
```

Comment:
We don't need to check the locked reason for the EditScope, because if the EditScope is locked, then that will be picked up by `nonEditableReason( valuePlug() )`.